urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/filehandler/IkarosExampleHandler.java

```diff
@@ -0,0 +1,283 @@
+	for( int i = 0; i < headers.size(); i++ )
+	    data += "   " + headers.get(i) + "\n";
+
+	data +=
+	    "Request Method:          " + headers.getRequestMethod() + "\n" +
+	    "Request Version:         " + headers.getRequestVersion() + "\n" +
+	    "Request Protocol:        " + headers.getRequestProtocol() + "\n" +
+	    "Request URI:             " + headers.getRequestURI() + "\n" +
+	    "POST data available:     " + (postData!=null) + "\n" +
+	    "\n" +
+	    "Session:                 " + session + "\n" +
+	    "Testvalue:               " + session.get("TEST") + "\n" +
+	    "                         (this value is null on first call or if your session expired)\n" +
+	    "\n"+
+	    makeSingleLineBox( "Note that this is a TCP/HTTP session, _not_ a browser session! " ) + "\n" + 
+	    "\n" +
+	    "\n" +
+	    makeFancyLineBox(  "      Did I mention that I love to make fancy text boxes?      " ) + "\n" +
+	    "\n" +
+	    "\n";
+
+	session.put( "TEST", new ikrs.typesystem.BasicStringType("123456") );
+
+
+	ByteArrayResource resource = new ByteArrayResource( this.getHTTPHandler(),
+							    this.getLogger(),
+							    data.getBytes(),
+							    false   // no need to use fair locks
+							    );
+
+	// I want this output to be displayed as plain text.
+	// One other posible way would be
+	//   MIMEType mimeType = new MIMEType( "text/plain" );
+	MIMEType mimeType = MIMEType.getByFileExtension( "txt" );
+	resource.getMetaData().setMIMEType( mimeType );
+	resource.getMetaData().setCharsetName( java.nio.charset.StandardCharsets.UTF_8.name() );
+		
+	return resource;
+    }
+    //--- END -------------------------- FileHandler implementation ------------------------------
+
+
+    private String makeDoubleLineBox( String line ) {
+
+	line = " " + line + " ";
+	StringBuffer b = new StringBuffer();
+
+	// Characters in US-ASCII (extended)
+	// DOES NOT WORK!
+	/*
+	b.append( (char)201 );  // left upper corner
+	for( int i = 0; i < line.length(); i++ )
+	    b.append( (char)205 );  // double horizonal line
+	b.append( (char)187 );  // right upper corner
+
+	b.append( (char)186 );  // double vertical line
+	b.append( line );
+	b.append( (char)186 );
+
+	b.append( (char)200 );  // left lower corner
+	for( int i = 0; i < line.length(); i++ )
+	    b.append( (char)205 );  // double horizonal line
+	b.append( (char)188 );  // right lower corner
+	*/
+	
+	// Characters in Unicode
+	// Does work :)
+	b.append( '\u2554' );  // left upper corner
+	for( int i = 0; i < line.length(); i++ )
+	    b.append( '\u2550' );  // double horizonal line
+	b.append( '\u2557' );  // right upper corner
+	b.append( "\n" );
+
+	b.append( '\u2551' );  // double vertical line
+	b.append( line );
+	b.append( '\u2551' );
+	b.append( "\n" );
+
+	b.append( '\u255A' );  // left lower corner
+	for( int i = 0; i < line.length(); i++ )
+	    b.append( '\u2550' );  
+	b.append( '\u255D' );  // right lower corner
+	//b.append( "\n" );
+
+	return b.toString();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 179:219, 225:247.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.